src/Frontend/Page/Idea.hs

```diff
@@ -104,6 +107,13 @@ data ReportComment = ReportComment Comment
 
 instance Page ReportComment where
 
+data EditComment
+    = EditComment Idea Comment
```

Comment:
you don't need `Idea`, because is part of the `CommentKey` that's part of the `Comment`.  I think.